simd/src/Kokkos_Simd_AVX512.hpp

```diff
@@ -0,0 +1,928 @@
+      return __mmask8(std::int16_t(1 << m_lane));
+    }
+
+   public:
+    KOKKOS_HOST_FORCEINLINE_FUNCTION reference(__mmask8& mask_arg, int lane_arg)
+        : m_mask(mask_arg), m_lane(lane_arg) {}
+    KOKKOS_HOST_FORCEINLINE_FUNCTION void operator=(bool value) const {
+      if (value) {
+        m_mask |= bit_mask();
+      } else {
+        m_mask &= ~bit_mask();
+      }
+    }
+    KOKKOS_HOST_FORCEINLINE_FUNCTION operator bool() const {
+      return (m_mask & bit_mask()) != 0;
+    }
+  };
+  using value_type                             = bool;
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd_mask() = default;
+  KOKKOS_HOST_FORCEINLINE_FUNCTION explicit simd_mask(value_type value)
+      : m_value(-std::int16_t(value)) {}
+  template <class U>
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd_mask(
+      simd_mask<U, simd_abi::avx512_fixed_size<8>> const& other)
+      : m_value(static_cast<__mmask8>(other)) {}
+  KOKKOS_HOST_FORCEINLINE_FUNCTION static constexpr std::size_t size() {
+    return 8;
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION constexpr explicit simd_mask(
+      __mmask8 const& value_in)
+      : m_value(value_in) {}
+  KOKKOS_HOST_FORCEINLINE_FUNCTION constexpr explicit operator __mmask8()
+      const {
+    return m_value;
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION reference operator[](std::size_t i) {
+    return reference(m_value, int(i));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION value_type operator[](std::size_t i) const {
+    return static_cast<value_type>(reference(m_value, int(i)));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd_mask
+  operator||(simd_mask const& other) const {
+    return simd_mask(_kor_mask8(m_value, other.m_value));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd_mask
+  operator&&(simd_mask const& other) const {
+    return simd_mask(_kand_mask8(m_value, other.m_value));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION simd_mask operator!() const {
+    static const __mmask8 true_value(static_cast<__mmask8>(simd_mask(true)));
+    return simd_mask(_kxor_mask8(true_value, m_value));
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION bool operator==(
+      simd_mask const& other) const {
+    return m_value == other.m_value;
+  }
+  KOKKOS_HOST_FORCEINLINE_FUNCTION bool operator!=(
+      simd_mask const& other) const {
+    return m_value != other.m_value;
+  }
+};
+
+template <class T>
+[[nodiscard]] KOKKOS_HOST_FORCEINLINE_FUNCTION bool all_of(
+    simd_mask<T, simd_abi::avx512_fixed_size<8>> const& a) {
+  static const __mmask16 false_value(-std::int16_t(false));
+  const __mmask16 a_value(0xFF00 | static_cast<__mmask8>(a));
+  return _kortestc_mask16_u8(a_value, false_value);
+}
+
+template <class T>
+[[nodiscard]] KOKKOS_HOST_FORCEINLINE_FUNCTION bool any_of(
+    simd_mask<T, simd_abi::avx512_fixed_size<8>> const& a) {
+  static const __mmask16 false_value(-std::int16_t(false));
+  const __mmask16 a_value(0x0000 | static_cast<__mmask8>(a));
+  return !_kortestc_mask16_u8(~a_value, false_value);
```

Comment:
Why are you using `__mmask16` here?

Reply:
Good catch. I think this was done by the STK team to support Xeon Phi where AVX512DQ is not provided (apparently `kortestc_mask8_u8` is only provided with AVX512DQ), but I guess I require AVX512DQ in this implementation so I should be able to use the mask8 one